index.js

```diff
@@ -147,18 +147,29 @@ function Metric(namespace, units, defaultDimensions, options) {
  * Publish this data to Cloudwatch
  * @param {Integer|Long} value          Data point to submit
  * @param {String} namespace            Name of the metric
- * @param {Array} additionalDimensions  Array of additional CloudWatch metric dimensions. See
+ * @param {String} units                CloudWatch units
+ * @param {Array} [additionalDimensions]  Array of additional CloudWatch metric dimensions. See
  * http://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_Dimension.html for details.
  */
-Metric.prototype.put = function(value, metricName, additionalDimensions) {
+Metric.prototype.put = function(...args) {
+  if (args.length === 3) {
+    const units = Array.isArray(args[2]) ? this.units : args[2];
+    const additionalDimensions = Array.isArray(args[2]) ? args[2] : [];
+    return this._put(args[0], args[1], units, additionalDimensions);
```

Comment:
Any reason not to destructure the first arguments of `args` as we do below (i.e. `const [value, metricName] = args;`)?